flocker/node/_deploy.py

```diff
@@ -930,15 +932,11 @@ def find_dataset_changes(uuid, current_state, desired_state):
                     going.add(DatasetHandoff(
                         dataset=dataset, hostname=hostname))
 
-    # Look at each dataset that is going to be hosted on this node.  If it
-    # was running somewhere else, we want that dataset to be in `coming`.
-    coming_dataset_ids = local_desired_dataset_ids.intersection(
-        remote_current_dataset_ids)
-    coming = set(dataset for dataset in local_desired_datasets
-                 if dataset.dataset_id in coming_dataset_ids)
-
```

Comment:
Mostly, yeah. Although originally I thought deleting that code would suffice, until I reached the creation code, and figured I may as well leave that improvement in.